application/models/user_fixtures_test.go

```diff
@@ -0,0 +1,276 @@
+
+	// Load User test fixtures
+	users := Users{
+		{
+			Email:     fmt.Sprintf("user1-%s@example.com", unique),
+			FirstName: "Eager",
+			LastName:  "User",
+			Nickname:  fmt.Sprintf("Eager User %s", unique),
+			Uuid:      domain.GetUuid(),
+		},
+		{
+			Email:     fmt.Sprintf("user2-%s@example.com", unique),
+			FirstName: "Lazy",
+			LastName:  "User",
+			Nickname:  fmt.Sprintf("Lazy User %s", unique),
+			Uuid:      domain.GetUuid(),
+		},
+	}
+	for i := range users {
+		if err := ms.DB.Create(&users[i]); err != nil {
+			t.Errorf("could not create test user %v ... %v", users[i], err)
+			t.FailNow()
+		}
+	}
+
+	// Load UserOrganization test fixtures
+	userOrgs := UserOrganizations{
+		{
+			OrganizationID: org.ID,
+			UserID:         users[0].ID,
+			AuthID:         users[0].Email,
+			AuthEmail:      users[0].Email,
+		},
+		{
+			OrganizationID: org.ID,
+			UserID:         users[1].ID,
+			AuthID:         users[1].Email,
+			AuthEmail:      users[1].Email,
+		},
+	}
+	for i := range userOrgs {
+		if err := ms.DB.Create(&userOrgs[i]); err != nil {
+			t.Errorf("could not create test user org ... %v. uo = %+v", err, userOrgs[i])
+			t.FailNow()
+		}
+	}
+
+	// Each user has a request and is a provider on the other user's post
+	posts := Posts{
+		{
+			CreatedByID:    users[0].ID,
+			OrganizationID: org.ID,
+			Type:           PostTypeRequest,
+			Title:          "Open Request 0",
+			Size:           PostSizeMedium,
+			Status:         PostStatusOpen,
+			Uuid:           domain.GetUuid(),
+			ProviderID:     nulls.NewInt(users[1].ID),
+		},
+		{
+			CreatedByID:    users[1].ID,
+			OrganizationID: org.ID,
+			Type:           PostTypeRequest,
+			Title:          "Committed Request 1",
+			Size:           PostSizeMedium,
+			Status:         PostStatusOpen,
+			Uuid:           domain.GetUuid(),
+			ProviderID:     nulls.NewInt(users[0].ID),
+		},
+	}
+
+	for i := range posts {
+		createFixture(ms, &posts[i])
+	}
+
+	threads := []Thread{
+		{
+			Uuid:   domain.GetUuid(),
+			PostID: posts[0].ID,
+		},
+		{
+			Uuid:   domain.GetUuid(),
+			PostID: posts[1].ID,
+		},
+	}
+
+	for i := range threads {
+		createFixture(ms, &threads[i])
+	}
+
+	tnow := time.Now()
+	oldTime := tnow.Add(-time.Duration(time.Hour))
+	futureTime := tnow.Add(time.Duration(time.Hour))
+
+	// One thread per post with 2 users per thread
+	threadParticipants := []ThreadParticipant{
+		{
+			ThreadID:     threads[0].ID,
+			UserID:       posts[0].CreatedByID,
+			LastViewedAt: futureTime,
+		},
+		{
+			ThreadID:     threads[0].ID,
+			UserID:       posts[0].ProviderID.Int,
+			LastViewedAt: oldTime,
+		},
+		{
+			ThreadID:     threads[1].ID,
+			UserID:       posts[1].CreatedByID,
+			LastViewedAt: oldTime,
+		},
+		{
+			ThreadID:     threads[1].ID,
+			UserID:       posts[1].ProviderID.Int,
+			LastViewedAt: futureTime,
+		},
+	}
+
+	for i := range threadParticipants {
+		createFixture(ms, &threadParticipants[i])
+	}
+
+	// I can't seem to give them custom times
```

Comment:
I had to do raw SQL inserts in `job_fixtures_test.go`. e.g.:
```go
		err := js.DB.RawQuery(`INSERT INTO messages (content, created_at, sent_by_id, thread_id, updated_at, uuid)
			VALUES (?, ?, ?, ?, ?, ?)`, m.Content, m.UpdatedAt, m.SentByID, m.ThreadID, m.UpdatedAt, m.Uuid).Exec()
```
It seems like you solved this by using future `last_viewed_at` times. That's fine with me.

Reply:
It's a shame they don't let you force the CreatedAt value.  (Maybe they do give a way to make that happen?)